doc/nrf/app_dev/device_guides/hpf/index.rst

```diff
@@ -0,0 +1,23 @@
+================================
+High Performance Framework (HPF)
+================================
```

Comment:
Please provide the link to a page on Confluence or a JIRA issue that describes the idea behind adding these docs, so that maintainers and requirements can be checked and if they meet [guidelines for doc sections](https://nordicsemi.atlassian.net/wiki/spaces/TECHDOC/pages/649694362/Guidelines+for+doc+sections+in+the+NCS)